janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -989,6 +990,70 @@ public void testConditionalIndexing() {
                 numV / strings.length, new boolean[]{false, false}, weight, Order.ASC);
     }
 
+    @Test
+    public void testGraphCentricQueryProfiling() {
+        final PropertyKey name = makeKey("name", String.class);
+        final PropertyKey prop = makeKey("prop", Integer.class);
+        final JanusGraphIndex mixedIndex = mgmt.buildIndex("mixed", Vertex.class)
+            .addKey(name, Mapping.STRING.asParameter()).addKey(prop).buildMixedIndex(INDEX);
+        finishSchema();
+
+        tx.addVertex("name", "bob", "prop", 100);
+        tx.commit();
+
+        // satisfied by a single graph-centric query which is satisfied by a single mixed index query
+        if (indexFeatures.supportNotQueryNormalForm()) {
+            newTx();
+            Metrics mMixedOr = tx.traversal().V().or(__.has("name", "bob"), __.has("prop", 100))
+                .profile().next().getMetrics(0);
+            assertEquals("Or(JanusGraphStep([],[name.eq(bob)]),JanusGraphStep([],[prop.eq(100)]))", mMixedOr.getName());
```

Comment:
That hard coded string is not very robust in case the `toString()` method for a traversal ever changes. Instead, you could initialize the traversal without the `profile()` step, save it's string representation and then apply the `profile()` step and perform the check.

Reply:
I intended to make it a change detector, to prevent unintentional changes to traverser/profiler.